lib/liquid/block_body.rb

```diff
@@ -13,8 +14,46 @@ def initialize
       @blank = true
     end
 
-    def parse(tokenizer, parse_context)
+    def parse(tokenizer, parse_context, &block)
       parse_context.line_number = tokenizer.line_number
+
+      if tokenizer.for_liquid_tag
+        parse_for_liquid_tag(tokenizer, parse_context, &block)
+      else
+        parse_for_document(tokenizer, parse_context, &block)
+      end
+    end
+
+    private def parse_for_liquid_tag(tokenizer, parse_context)
+      while token = tokenizer.shift
+        case
+        when token.empty?
+          # pass
+        else
+          unless token =~ LiquidTagToken
+            next if token =~ WhitespaceOrNothing
+            # line isn't empty but didn't match tag syntax, yield and let the
+            # caller raise a syntax error
+            return yield token, token
```

Comment:
This implementation is weird and should be debated. We can hit this case, for example, with:

```liquid
{% liquid
  !!! this is not a tag
%}
```

The outcome is a `syntax error (line 2): Unknown tag '!!! this is not a tag'`, which is what I was going for. The weird part about the implementation is that the block is supposed to be passed `tag_name, markup`, but since we couldn't parse the tag name, I just passed `token` for both. Alternatively, since we know at this point we don't have a valid `endif` or `endfor` or another such closing tag, we could just raise here, which is much less confusing to me. Thoughts?

Reply:
```liquid
{% liquid
  !!! this is not a tag
%}
```

seems like it should be handled the same way as

```liquid
{% !!! this is not a tag %}
```

which raises instead of yielding.

Currently it uses `raise_missing_tag_terminator`, but it could use a more appropriate error message in this case where tag has been terminated with `%}`.

Reply:
Let's raise then, it's definitely easier to understand.

Perhaps we should also fix handling for `{% !!! this is not a tag %}`. `Liquid syntax error: Tag '{% !!! this is not a tag %}' was not properly terminated with regexp: /\%\}/` is the wrong error message.